ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HivePointLookupOptimizerRule.java

```diff
@@ -189,115 +189,121 @@ public RexNode analyzeRexNode(RexBuilder rexBuilder, RexNode condition) {
     return newCondition;
   }
 
-  /**
-   * Transforms inequality candidates into [NOT] BETWEEN calls.
-   *
-   */
-  protected static class RexTransformIntoBetween extends RexShuttle {
-    private final RexBuilder rexBuilder;
+  public static class DiGraph<V, E> {
```

Comment:
Left the comment in other class but I was thinking that it may be a good idea to promote this to top class (at least until we replace it by any other library version as we were discussing).